msgp/elsize_tinygo.go

```diff
@@ -0,0 +1,129 @@
+// +build tinygo
+
+package msgp
+
+// for tinygo, we use a simple/slow function with a switch
+// statement - doesn't require any heap alloc, moves the space
+// requirements into code instad of ram
+
+func getBytespec(v byte) bytespec {
+
+	// put this all in a big switch and let the compiler figure it out
+	switch {
```

Comment:
For the switch here, let's do `switch v { case mnil: ... }` and then have a second switch for the cases that don't match constants.

Reply:
fixed